pom.xml

```diff
@@ -42,5 +48,59 @@
             <scope>test</scope>
         </dependency>
     </dependencies>
-
+    
+    <distributionManagement>
+        <repository>
+            <id>internal.repo</id>
+            <name>Temporary Staging Repository</name>
+            <url>file://${project.build.directory}/mvn-repo</url>
+        </repository>
+    </distributionManagement>
+    
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-deploy-plugin</artifactId>
+                <version>2.8.2</version>
+                <configuration>
+                    <altDeploymentRepository>internal.repo::default::file://${project.build.directory}/mvn-repo</altDeploymentRepository>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>com.github.github</groupId>
+                <artifactId>site-maven-plugin</artifactId>
+                <version>0.12</version>
+                <configuration>
+                    <!-- git commit message -->
+                    <message>Maven artifacts for ${project.version}</message>
+                    <!-- disable webpage processing -->
+                    <noJekyll>true</noJekyll>
+                    <!-- matches distribution management repository url above -->
+                    <outputDirectory>${project.build.directory}/mvn-repo</outputDirectory>
+                    <!-- remote branch name -->
+                    <branch>refs/heads/mvn-repo</branch>
+                    <!-- If you remove this then the old artifact will be removed and new 
+                    one will replace. But with the merge tag you can just release by changing 
+                    the version -->
+                    <merge>true</merge>
+                    <includes>
+                        <include>**/*</include>
+                    </includes>
+                    <!-- github repo name -->
+                    <repositoryName>mangaeden-client</repositoryName>
+                    <!-- github username -->
+                    <repositoryOwner>zygimantus</repositoryOwner>
```

Comment:
Is there a way to make this work for my repo too? Maybe just make it configurable?

Reply:
Change the repositoryOwner to your name and you also need github.global.server property to your Maven's server.xml file, etc. like this:

<servers>
  <server>
    <id>github</id>
    <username>jwang47</username>
    <password>****************</password>
  </server>
</servers>